pkg/controller/certificates/certificate_controller.go

```diff
@@ -221,3 +230,24 @@ func (cc *CertificateController) maybeSignCertificate(key string) error {
 	_, err = cc.kubeClient.Certificates().CertificateSigningRequests().UpdateStatus(csr)
 	return err
 }
+
+func (cc *CertificateController) sign(csr *certificates.CertificateSigningRequest) ([]byte, error) {
+	cc.signerLock.Lock()
+	defer func() {
```

Comment:
This is awkward.  Why not just make `signer` an interface and pass in the CSR to it and let it decide?  This isn't really something the cert controller should be doing.